crates/oxc_transformer/src/typescript/diagnostics.rs

```diff
@@ -0,0 +1,10 @@
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_span::Span;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("`import lib = require(...);` is only supported when compiling modules to CommonJS.\nPlease consider using `import lib from '...';` alongside Typescript's --allowSyntheticDefaultImports option, or add @babel/plugin-transform-modules-commonjs to your Babel config.")]
```

Comment:
@Dunqing @Boshen do we want to reference babel here?

Reply:
I think that until oxc has the modules-commonjs plugin, we can make its errors consistent with babel for the time being